doc/fileformat/cable_cell.rst

```diff
@@ -0,0 +1,429 @@
+   follows the envelope. A non-zero ``freq`` implies the current is sinusoidal with that frequency and amplitude
+   that varies according to the envelope. (More information about current clamps can be found
+   :ref:`here <cablecell-stimuli>`).
+   For example:
+
+   .. code::
+
+      (current-clamp (envelope (0 10) (50 10) (50 0)) 40)
+
+   This expression describes a sinusoidal current with amplitude 10nA and frequency 40Hz and that lasts
+   from t = 0ms to t = 50ms, finally leaving the current at 0nA (final amplitude in the envelope).
+
+.. label:: (threshold-detector val:real).
+
+   This describes a *threshold-detector* object with value ``val`` (unit [mV]]).
+
+.. label:: (gap-junction-site)
+
+   This describes a *gap-junction-site*.
+
+*Paintable* and *placeable* properties and dynamics are placed on regions and locsets respectively.
+Arbor provides many useful :ref:`region expressions <labels-region-expr>` and
+:ref:`locset expressions <labels-locset-expr>` which are explained in detail at the provided links.
+*Defaultable* properties and dynamics apply to an entire cell.
+
+.. label:: (paint reg:region prop:paintable)
+
+   This applies the painatble property ``prop`` to region ``reg``.
+   For example:
+
+   .. code:: lisp
+
+      (paint (tag 1) (membrane-capacitance 0.02))
+
+   This expression sets the membrane capacitance of the region tagged ``1`` to 0.02 F/m².
+
+
+.. label:: (place ls:locset prop:placeable)
+
+   This places the property ``prop`` on locset ``ls``.
+   For example:
+
+   .. code:: lisp
+
+      (place (locset "mylocset") (threshold-detector 10))
+
+   This expression places a 10 mV threshold detector on the locset labeled ``mylocset``.
+   (The definition of ``mylocset`` should be provided in a label dictionary associated
+   with the decor).
+
+.. label:: (default prop:defaultable)
+
+   This sets the property ``prop`` as default for the entire cell. (This default property can be overridden on region
+   using a ``paint`` expression).
+   For example:
+
+   .. code:: lisp
+
+      (default (membrane-potential -65))
+
+   This expression sets the default membrane potential of the cell to -65 mV.
+
+Any number of paint, place and default expressions can be used to create a decor as follows:
+
+.. label:: (decor [...def:paint/place/default])
+
+   This describes a decor object with zero or more paint, place or default expressions in any order.
+   For example:
+
+   .. code:: lisp
+
+      (decor
+        (default (membrane-potential -55.000000))
+        (paint (region "custom") (temperature-kelvin 270))
+        (paint (region "soma") (membrane-potential -50.000000))
+        (paint (all) (mechanism "pas"))
+        (paint (tag 4) (mechanism "Ih" ("gbar" 0.001)))
+        (place (locset "root") (mechanism "expsyn"))
+        (place (terminal) (gap-junction-site)))
+
+Morphology
+----------
+
+The morphology of a cable cell can be described in terms of points, tagged segments and branches.
```

Comment:
The difference between segment tree and morphology is a bit subtle, and gets more confusing if we points out the segments in a morphology. Does the format in fact decribe a morphology or segment_tree?

Reply:
Technically, we build a segment tree and create a morphology from it. I know there's a subtle difference in the `morphology` and `segment_tree` classes, but we create morphologies from segment trees all the time in the docs and the examples; so I don't think using `morphology` here will cause confusion, in fact I think `segment-tree` would be more confusing.

Reply:
We can remove it from the concepts, maybe.